src/utils/blockchain.ts

```diff
@@ -134,6 +154,120 @@ const getRawTransaction = async (asset: string, txid: string) => {
     return await fetchBlockExplorer<string>(asset, `/tx/${txid}/hex`);
 };
 
+export const getBlockTipHeight = async (asset: string) => {
+    const height = await fetchBlockExplorer<string>(
+        asset,
+        "/blocks/tip/height",
+    );
+
+    if (!Number.isFinite(Number(height))) {
+        throw new Error(
+            `invalid block tip height for asset ${asset}: ${height}`,
+        );
+    }
+
+    return height;
+};
+
+const getEsploraFeeEstimations = async (asset: string) => {
+    const { opts, requestTimeout } = constructRequestOptions();
+    try {
+        const esploraApi = getExplorerApi(asset, Explorer.Esplora);
+
+        if (!esploraApi) {
+            throw new Error(`no esplora API found for asset ${asset}`);
+        }
+
+        const endpoint = `${chooseUrl(esploraApi)}/fee-estimates`;
+
+        const res = await fetch(endpoint, opts);
+
+        if (!res.ok) {
+            await handleResponseError(res);
+        }
+
+        return (await res.json()) as Record<string, number>;
+    } finally {
+        clearTimeout(requestTimeout);
+    }
+};
+
+const getMempoolFeeEstimations = async (asset: string) => {
+    type MempoolFeeEstimation = Record<
+        "fastestFee" | "halfHourFee" | "hourFee" | "economyFee" | "minimumFee",
+        number
+    >;
+
+    const { opts, requestTimeout } = constructRequestOptions();
+    try {
+        const mempoolApi = getExplorerApi(asset, Explorer.Mempool);
+
+        if (!mempoolApi) {
+            throw new Error(`no mempool API found for asset ${asset}`);
+        }
+
+        const endpoint = `${chooseUrl(mempoolApi)}/v1/fees/recommended`;
+
+        const res = await fetch(endpoint, opts);
+
+        if (!res.ok) {
+            await handleResponseError(res);
+        }
+
+        return (await res.json()) as MempoolFeeEstimation;
+    } finally {
+        clearTimeout(requestTimeout);
+    }
+};
+
+export const getExplorerFeeEstimations = async (asset: string) => {
+    const handleUnexpectedResponse = (
+        feeEstimations: Record<string, number>,
+    ) => {
+        throw new Error(
+            `unexpected fee estimation response for ${asset}: ${JSON.stringify(
+                feeEstimations,
+            )}`,
+        );
+    };
+
+    // we don't have mempool on regtest
+    if (config.network !== "regtest") {
```

Comment:
Why hardcode that. Just don't try to use mempool, when we don't have a mempool instance configured